presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergAbstractMetadata.java

```diff
@@ -258,59 +258,52 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
 
         Table icebergTable = getIcebergTable(session, tableHandle.getSchemaTableName());
         validateTableMode(session, icebergTable);
-
+        List<ColumnHandle> partitionColumns = ImmutableList.copyOf(icebergTableLayoutHandle.getPartitionColumns());
         if (!isPushdownFilterEnabled(session)) {
-            return new ConnectorTableLayout(handle);
-        }
-
-        if (!icebergTableLayoutHandle.getPartitions().isPresent()) {
             return new ConnectorTableLayout(
                     icebergTableLayoutHandle,
                     Optional.empty(),
-                    TupleDomain.none(),
+                    icebergTableLayoutHandle.getPartitionColumnPredicate(),
```

Comment:
Nice job, it's pretty good to fill partition predicate constraint into layout for shown up. I am investigating and confirming in my local branch about totally discard the predicate in IcebergTableHandle. So that the responsibilities of `IcebergTableHandle`, `IcebergTableLayoutHandle` and `ConnectorTableLayout` could be clarified more clearer, and the behaviors could be more consistent between `filter_push_down` on and off.

Reply:
I agree with making that change. Having a predicate specified in multiple places currently is quite confusing